### ISSUE TITLE
Combining Vars into one Rx without mapping

### ISSUE THREAD

**Issue by cornerman** [OlivierBlanvillain/monadic-html#40]
While combining two reactive `Var` into one `Rx`, I noticed that this behaves differently depending on whether the inner variable in the outer flatMap is mapped or not.

The following example uses two variables `var1` and `var2`. Both are combined into `rx1` and `rx2` using `flatMap`:

```
  val var1 = Var(1)
  val var2 = Var(2)

  val rx1 = var1.flatMap(_ => var2)
  val rx2 = var1.flatMap(_ => var2.map(id => id))

  rx1.foreach(_ => println("rx1 triggers"))
  rx2.foreach(_ => println("rx2 triggers"))

  var1 := -1
  var2 := -2
```

I would expect both, `rx1` and `rx2`, to behave exactly the same, but it prints the following:

```
rx1 triggers
rx2 triggers
rx2 triggers
rx1 triggers
rx2 triggers
```

The first two triggers come from the initialization and the last two come from changing `var2`. Both is fine. But, the third println comes from chaning `var1` and only triggers `rx2` and is missing `rx1`. Any ideas what is happening here?

**Comment by OlivierBlanvillain** [comment]
That's a bug. This is an optimization leaking, I think I'm gonna need to remove this one...

The optimization is basically [this line](https://github.com/OlivierBlanvillain/monadic-html/blob/de78d4b147e39f5fdec300cce4c2b68e3596bf3a/monadic-rx/shared/src/main/scala/mhtml/rx.scala#L27). Since the output of `_ => var2` is stable wrt referential equality the optimization decides that the output does not need to be trigger whenever `var1` updates, whereas with `_ => var2.map(id => id)` the returned `Rx` is new every time (the `.map` instantiates a new one!), and the output is systematically updates.

**Comment by OlivierBlanvillain** [comment]
Fixed with the rework in #39
